src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs

```diff
@@ -506,7 +498,7 @@ protected override bool ReplacementChangesSemanticsForNodeLanguageSpecific(Synta
             }
             else if (currentOriginalNode.Kind() == SyntaxKind.ImplicitArrayCreationExpression)
             {
-                return !TypesAreCompatible((ImplicitArrayCreationExpressionSyntax)currentOriginalNode, (ImplicitArrayCreationExpressionSyntax)currentReplacedNode);
+                return !TypesAreCompatible((ExpressionSyntax)currentOriginalNode, (ExpressionSyntax)currentReplacedNode);
```

Comment:
incorrect logic before. the point of this analyzer is that the old construct and new construct might not match, but semantics shoudl still be preserved.  so, in this case, the new construct just needs to be an expression, not an implicit array like before.